### ISSUE TITLE
audio engine crash when using with touch instrument

### ISSUE THREAD

**Issue by kalaniseguin** [git-moss/Push4Bitwig#194]
So after a lot of testing, I've concluded it's at least a possibility it is this script which is causing my issue, so I figure I'd post this here, while also reaching out to Bitwig. 
What happens is when I am playing the Push while simultaneously using the touch instrument built into bitwig, the Audio engine crashes, or the audio hangs until I press the bitwig killswitch, in which case turning it back on either brings the project back to working, or it doesn't, in which case the only solution is quit and reopen.
This consistently happens every time now, where it never happened before.  I recall it starting around mid December, but unfortunately can't recall whether it was a script update or bitwig update that was the exact moment it started.
It does not happen when using a keyboard midi controller and the touch instrument simultaneously, and increasing the buffer size makes no difference.
One clue it may be the script is the fact that the led's of the notes being held on the push prior to the crash stay illuminated until the reset happens, though of course that proves nothing. 
It may not seem like a widely seen bug, as I can't imagine a lot of people use bitwig in this way, but I have found the Push with touch instrument combination to be the most inspiring way of creating music currently, and I'd really like to be able to do it again.
I will be reaching out to bitwig regarding this as well, and of course have submitted crash reports each time it's happened.  
I sincerely hope this can be resolved, and thank you very much for all your hard work!

**Comment by kalaniseguin** [comment]
was using the most recent stable build, and bitwig 1.3.6, as well as the prior script and 1.3.5, all with the same results.

**Comment by git-moss** [comment]
Is this still with 1.3.8 RC 2?

**Comment by kalaniseguin** [comment]
just tested, and yes unfortunately it is...

**Comment by git-moss** [comment]
Since I also have a touchscreen I'd like to replicate this bug. Can you give a step-by-step description of how to make it crash?

**Comment by kalaniseguin** [comment]
Pretty easy, but maybe you can clarify if it's just me. Basically pull up an instrument track, even without a device loaded in I've had it happen, but load any device, polyphonic to get the purpose of my use case. Hold notes on push, and play a note on the touch instrument (have to be in tablet mode "play" window of course). It may require bending of the note on the screen, as that's typically what I'm doing. Most of the time, it happens with the very first touch screen note, but I did experience one time about a week ago where it took maybe 4 notes before it crashed. Let me know if you can replicate, and if not, it could have something to do with speck rally surface books? Though again, it didn't happen throughout November.

**Comment by git-moss** [comment]
I can replicate the problem. The audio engine did not crash but I got hanging notes and the script was dead. Even restarting the script and/or the Push did not help. A complete Bitwig restart was necessary. I tried to replicate it with Push and keyboard but it does not happen there.
Seems to be related to the touchscreen bitch bend.

**Comment by kalaniseguin** [comment]
Right. That's what it appears. Perhaps the fact that it is note independent, whereas the pitch bend from the ribbon on push would be universal like normal? Note this happens even when they are assigned to 2 different instruments on different tracks. Also, that it did not happen back in November. Something changed between then and now, as I've said, mid December I want to say, to cause this. I've sent countless crash reports, as it does crash my system often, though I have experienced hangs as well, which I described in my initial report. Anyway, hope it can get fixed!

**Comment by kalaniseguin** [comment]
I was thinking, I seem to recall back when I wasn't having this issue I may not have been seeing led feedback on the push when using the touch instrument, whereas now I do. IE if I just use the touch instrument I can see pads representing the notes light up on the push, but of course at the time I'm not using the push.  Is it possible the issue is related to midi feedback from the touch instrument to the push, in which case it's again because of the polyphonic pitch bend, and in that case would there be a way to "mute" those signals to the push?

**Comment by git-moss** [comment]
Still in 1.3.13.
